packages/scripts/src/admin-proposals/change-final-fee/0_Propose.ts

```diff
@@ -0,0 +1,175 @@
+import { tokensToUpdateFee } from "./common";
+
+interface AdminProposalTransaction {
+  to: string;
+  value: BigNumberish;
+  data: BytesLike;
+}
+
+const proposerWallet = "0x2bAaA41d155ad8a4126184950B31F50A1513cE25";
+
+// Env vars
+const NODE_URL_ENV = "NODE_URL_";
+
+async function main() {
+  const adminProposalTransactions: AdminProposalTransaction[] = [];
+
+  let proposerSigner: Signer;
+
+  if (!process.env.NEW_FINAL_FEE_USD) throw new Error("NEW_FINAL_FEE_USD is not set");
+  if (!process.env.NEW_FINAL_FEE_WETH) throw new Error("NEW_FINAL_FEE_WETH is not set");
+  if (!process.env.UMIP_NUMBER) throw new Error("UMIP_NUMBER is not set");
+
+  const umipNumber = Number(process.env.UMIP_NUMBER);
+
+  const newFinalFeeUSD = Number(process.env.NEW_FINAL_FEE_USD);
+  const newFinalFeeWeth = Number(process.env.NEW_FINAL_FEE_WETH);
+
+  const arbitrumParentMessenger = await getContractInstance<ParentMessengerBaseEthers>("Arbitrum_ParentMessenger");
+
+  const governorRootTunnel = await getContractInstance<GovernorRootTunnelEthers>("GovernorRootTunnel"); // for polygon
+  const governorHub = await getContractInstance<GovernorHubEthers>("GovernorHub"); // rest of l2
+
+  const l2Networks = ["polygon", "arbitrum", "optimism"];
+  const l2NetworksNumber = { polygon: 137, optimism: 10, arbitrum: 42161 };
+
+  if (process.env.GCKMS_WALLET) {
+    proposerSigner = ((await getGckmsSigner()) as Wallet).connect(hre.ethers.provider as Provider);
+    if (proposerWallet.toLowerCase() != (await proposerSigner.getAddress()).toLowerCase())
+      throw new Error("GCKMS wallet does not match proposer wallet");
+  } else {
+    proposerSigner = (await hre.ethers.getSigner(proposerWallet)) as Signer;
+  }
+
+  const votingToken = await getContractInstance<VotingTokenEthers>("VotingToken");
+
+  const proposer = await await getContractInstance<ProposerV2Ethers>("ProposerV2");
+  const store = await getContractInstance<StoreEthers>("Store");
+
+  // case mainnet
+  for (const token in tokensToUpdateFee) {
+    const provider = getRetryProvider(1) as Provider;
+
+    const isWeth = token == "WETH";
+    const newFinalFee = isWeth ? newFinalFeeWeth : newFinalFeeUSD;
+
+    const tokenAddress = tokensToUpdateFee[token as keyof typeof tokensToUpdateFee].mainnet;
+    const erc20 = await getContractInstanceWithProvider<ERC20Ethers>("ERC20", provider, tokenAddress);
+    const decimals = await erc20.decimals();
+
+    const setFinalFeeTx = await store.populateTransaction.setFinalFee(tokenAddress, {
+      rawValue: hre.ethers.utils.parseUnits(newFinalFee.toString(), decimals),
+    });
+    if (!setFinalFeeTx.data) throw "setFinalFeeTx.data is null";
+    adminProposalTransactions.push({ to: store.address, value: 0, data: setFinalFeeTx.data });
+  }
+
+  for (const i in l2Networks) {
+    const networkName = l2Networks[i];
+    const provider = getRetryProvider(l2NetworksNumber[networkName as keyof typeof l2NetworksNumber]) as Provider;
```

Comment:
nit: if you add "as const" to the end of the declaration of these l2Networks variables above, ts should infer that the elements of the array are all keys of the object (I think!)